src/nile/core/declare.py

```diff
@@ -42,6 +44,13 @@ def declare(
     logging.info(f"🧾 Transaction hash: {hex(tx_hash)}")
 
     deployments.register_class_hash(class_hash, network, alias)
+
+    if watch_mode is not None:
+        is_accepted = status(tx_hash, network, watch_mode)
+        if is_accepted.status == Status.REJECTED:
+            deployments.unregister(class_hash, network, alias, is_deployment=False)
```

Comment:
You guys are doing great, thank you for giving more and more depth to the PR!
Here, I'd like to say that the `Status` class itself was not meant to be exported (for the `[...] import Status, status` situation).

Instead, these line could be written as
 
```suggestion
        tx_status = status(tx_hash, network, watch_mode)
        if tx_status.status.is_rejected:
```
or even
```suggestion
        if status(tx_hash, network, watch_mode).status.is_rejected:
```
which allows to keep the implementation of is_rejected separate.